app/controllers/api/projects_controller.rb

```diff
@@ -23,23 +23,79 @@ def searchcode
   end
 
   def dependencies
-    version = if params[:version] == 'latest'
-                @project.versions.sort.first
-              else
-                @project.versions.find_by_number(params[:version])
-              end
-
-    raise ActiveRecord::RecordNotFound if version.nil?
-
-
+    version = query_version(@project, params[:version])
 
     project_json = ProjectSerializer.new(@project).as_json
     project_json[:dependencies] = map_dependencies(version.dependencies || [])
 
     render json: project_json
   end
 
+  def dependencies_only
```

Comment:
Instead of an all-new endpoint, what about just a query string parameter to adjust the returned data? That feels a little cleaner

Reply:
any thoughts on what it would be? the clean thing might be some general filter param but don’t want to make things complicated. `?subset=dependencies` as a hardcoded single filter mode?

Reply:
?subset=dependencies works for me for now. Let’s show it working and then can think about if we want to generalize it more

Reply:
?subset=dependencies works for me for now. Let’s show it working and then can think about if we want to generalize it more

Reply:
`/dependencies?subset=dependencies` was kind of a silly suggestion on my part so I changed it to `/dependencies?subset=minimum`